### ISSUE TITLE
Reinstalling HB fails

### ISSUE THREAD

**Issue by hairuo** [Homebrew/homebrew-cask#51550]
#### General troubleshooting steps

- [x] I have retried my command with `--force` and the issue is still present.
- [x ] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [x] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/a_cask_fails_to_install.md#checksum-does-not-match-error) error.
- [x] I ran `brew update-reset && brew update` and retried my command.
- [x] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [x] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).

#### Description of issue

I have listed my problem in #49726, but it was closed, and the suggestion is:
reinstalling HB

However, I have tried reinstalling HB and it fails, which has been described in #49726. So the problem is not solved and still closed.

I have no way but post an issue again here.

I don't know how HB works, could any one help me? thank you very much!

**Comment by vitorgalvao** [comment]
> I have no way but post an issue again here.

Yes, you do. You could have posted in the same issue as before. Closing does not prevent you from posting on an issue, as you know because you’ve done it. “reinstalling HB fails” doesn’t tell us anything. What failed? What were the steps? Where did it break? What was the output? What did you try?

**Comment by hairuo** [comment]
Hi @vitorgalvao , thanks for your patience and your chance for me to argue. I really don't know much about how HB works and what command means during the process. I'm sorry to post an new issue to emphasis the previous problem again just because I think the previous issue is not solved and closed. 

I was just puzzled by why  #49726 is closed since the problem is not solved.

As you said that "We’re not being able to reproduce, so something must’ve got messed up in your local setup. If you’re still encountering this, I suggest reinstalling HB."

However, in #49726 I have tried reinstalling HB and it still can't fix the problem I mentioned, by using 
```
$ brew uninstall --force brew-cask
```
to uninstall HB and it fails(I don't know why and how to uninstall HB in other way), So I tried to  use 
```
/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
```
again to hope it can be reinstalled and again it fails. 

So could you or anyone answer #49726 clearly how to uninstall HB completely, or what commands could be used?  Then I would accept #49726 to be closed completely without any disagreement.

I don't know what happened, and I don't know what you means "local setup", could someone help me to use HB again, I do love HB and hope it works again. Thank you for your attention.

**Comment by vitorgalvao** [comment]
> I was just puzzled by why #49726 is closed since the problem is not solved.

Because the problem is in your setup, not related to the normal behaviour of HBC.

> `brew uninstall --force brew-cask`

There’s your problem. That command does nothing. `brew-cask` is no longer a formula, but an integral part of Homebrew.

To uninstall HB (you might need to reinstall your packages afterwards, see the [FAQ](https://docs.brew.sh/FAQ) and search for `How do I uninstall Homebrew?`. You’ll find the command to run.
